Topics/Tech_Stacks/Introduction_to_Django_Templates.md

```diff
@@ -0,0 +1,193 @@
+
+We can also pass in lists and dictionaries. Using a `.` performs a lookup to access by index/attribute:
+```
+<li>{{ menu_items.0.name }}</li>
+```
+
+Assuming menu_items is a list of dictionaries, this would display the first dictionary's name value.
+
+### Built-in Tags
+DTL's built-in tags is a convenient way to allow for conditional statements, loops, loading static files, and [more](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/). Tags are characterized by the `%` symbol inside curly braces.
+
+#### Conditional Statements
+The syntax and [boolean operators](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#boolean-operators) for conditionals inside the tags are similar to Python, except it must end with an `{% endif %}` tag.:
+```
+{% if menu_item.is_special %}
+    <li class="special">{{ menu_item.name}}</li>
+{% else %}
+    <li class="special">{{ menu_item.name}}</li>
+{% endif %}
+```
+
+#### Loops
+We can loop through lists, dictionaries, and its nested variants with the familiar Python syntax. Like conditionals, it must end with an `{% endfor %}` tag:
+```
+{% for menu_item in menu_items %}
+ {% if menu_item.is_special %}
+    <li class="special">{{ menu_item.name }} {{ menu_item.price }}</li>
+  {% else %}
+    <li class="regular">{{ menu_item.name }} {{ menu_item.price }}</li>
+  {% endif %}
+{% endfor %}
+```
+
+#### Including CSS Files
+First, we can create a directory called `static` in the root `project` directory. Inside `static`, it is typical to have subdirectories for file types such as images, `.css`, and `.js` files:
+```
+project/
+└── static/
+    └── css/
+        └── styles.css
+    └── js/
+    └── img/
+└── app/
+    └── ...
+└── ...
+```
+With this setup, we have to configure `settings.py`:
+```
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/4.2/howto/static-files/
+STATIC_URL = 'static/'
+STATICFILES_DIRS = [ BASE_DIR / 'static', ]
+```
+
+`STATICFILES_DIRS = [ BASE_DIR / 'static', ]` tells Django to look in the `static` folder in the root directory. However, there are times where you may want to have `static` directories in each app directory, which can be configured accordingly.
+
+In the template, we must load the static files with the `{% load static %}` tag, and include a specific file using the `{% static %}` tag. Using the above file structure, we can include the `.css` file:
+```
+{% load static %}
+<link rel="stylesheet" href="{% static 'css/styles.css' %}">
+```
+Note that the `{% load static %}` tag must be used before any `{% static %}` tag, so it is generally placed at the top of the template.
+
+### Built-in Filters
```

Comment:
I think having this subsection before "Including CSS Files" would be better since you were talking about variables earlier

Reply:
The logical flow of introducing the syntax follows the way I listed them in the introductory paragraph "Basic Syntax". So it makes sense to me to leave it after the "Built-in Tags" section.

Reply:
I see, thanks for clarifying